editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/adopters/AbstractProjectTemplate.java

```diff
@@ -47,18 +48,18 @@ public boolean supportsDSL(CamelDSLType type) {
 	 * @return	true on success, otherwise false
 	 * @throws CoreException	on any error
 	 */
-	public final boolean create(IProject project, NewProjectMetaData projectMetaData) throws CoreException {
-		IProgressMonitor monitor = new NullProgressMonitor();
+	public final boolean create(IProject project, NewProjectMetaData projectMetaData, IProgressMonitor monitor) throws CoreException {
+		SubMonitor subMonitor = SubMonitor.convert(monitor, 100);
```

Comment:
I understand, but maybe something to consider is that I think most users are pretty used to the individual chunks of a progress monitor not even coming close to reflecting an actual time.  Meaning, for instance, given a monitor where it visually appears to have 3 equally sized tasks, each taking up a third of the bar, the first and second may for instance be very quick, while the third takes in actuality 90% of the time.  Even that, seems like it might make more sense to NOT guess at some “in the ballpark” proportion and just set everything to have the same number of ticks.  Just something to consider.

> On Jul 25, 2016, at 10:47 AM, Aurélien Pupier <notifications@github.com> wrote:
> 
> In editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/adopters/AbstractProjectTemplate.java <https://github.com/fusesource/fuseide/pull/601#discussion_r72089438>:
> 
> > @@ -47,18 +48,18 @@ public boolean supportsDSL(CamelDSLType type) {
> >  	 * @return	true on success, otherwise false
> >  	 * @throws CoreException	on any error
> >  	 */
> > -	public final boolean create(IProject project, NewProjectMetaData projectMetaData) throws CoreException {
> > -		IProgressMonitor monitor = new NullProgressMonitor();
> > +	public final boolean create(IProject project, NewProjectMetaData projectMetaData, IProgressMonitor monitor) throws CoreException {
> > +		SubMonitor subMonitor = SubMonitor.convert(monitor, 100);
> The division of work should reflect the proportion.
> In other places, there are often long-running tasks for which I wasn't able/don't take time to guess/determine the proportions, so I just put '1' for each of them.
> I tried to be more precise at this place
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub <https://github.com/fusesource/fuseide/pull/601/files/db365b4c04b1928539754bfbf7bd474c2abced59#r72089438>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAarXfYh9kB8JeOrv8W-L84MhvU0Ncsdks5qZNp0gaJpZM4JT9y3>.
>